webui/react/src/e2e/models/BaseComponent.ts

```diff
@@ -0,0 +1,83 @@
+import { type Locator } from '@playwright/test';
+import { BasePage } from './BasePage';
+
+/**
+ * Used in the constructor for BaseComponent
+ */
+export interface BaseComponentProps {
+  parent: BasePage | BaseComponent
+  selector?: string
+}
+
+/**
+ * Returns the representation of a Component.
+ * 
+ * @remarks
+ * This constructor is a base class for any component in src/components/.
+ * 
+ * @param {Object} obj
+ * @param {BasePage | BaseComponent} obj.parent - The parent used to locate this BaseComponent
+ * @param {string} [obj.selector] - Used instead of `defaultSelector`
+ */
+export class BaseComponent {
+  readonly defaultSelector: undefined | string;
+
+  readonly #selector: string;
+  protected readonly _parent: BasePage | BaseComponent;
```

Comment:
Is there any way we can bring this down to one base class? using | basically means this needs double handling everywhere going forward. It'd be nice if there was a single root class.
Of course, if we actually DO always need to handle this in two ways this could be a good guardrail. But in that case maybe there should be two parent fields? Just brainstorming a bit here mostly.